lib/utils/fs.go

```diff
@@ -241,32 +241,97 @@ func FSTryReadLockTimeout(ctx context.Context, filePath string, timeout time.Dur
 	return fileLock.Unlock, nil
 }
 
+// RemoveAllSecure is similar to [os.RemoveAll] but leverages [RemoveSecure] to delete files.
+func RemoveAllSecure(path string) error {
+	if path == "" {
+		// match behavior from os.RemoveAll
+		return nil
+	}
+	// Match os.RemoveAll protections in not permitting removal of "." directories
+	// This check comes directly from https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/os/removeall_at.go;l=24
+	if path == "." || (len(path) >= 2 && path[len(path)-1] == '.' && os.IsPathSeparator(path[len(path)-2])) {
+		return &os.PathError{Op: "RemoveAllSecure", Path: path, Err: syscall.EINVAL} // error type matches os.RemoveAll
+	}
+
+	info, err := os.Lstat(path)
+	switch {
+	case err != nil && os.IsNotExist(err):
+		return nil
+	case err != nil:
+		return trace.ConvertSystemError(err)
+	case !info.IsDir():
+		return removeSecure(path, info)
+	}
+	var removeErrors []error
+	files, err := os.ReadDir(path)
+	if err != nil {
+		// Don't fail fast, allow removal at end to be attempted.
+		removeErrors = []error{err}
```

Comment:
```suggestion
		removeErrors = append(removeErrors, err)
```

Append to the slice we already have rather than overwrite it with a new one?